packages/api/src/domain/medical/document-query.ts

```diff
@@ -19,41 +17,22 @@ export type DocumentQueryProgress = {
 export const convertResult = ["success", "failed"] as const;
 export type ConvertResult = (typeof convertResult)[number];
 
-export const calculateConversionProgress = ({
-  patient,
-  convertResult,
-}: {
-  patient: Pick<Patient, "id" | "cxId">;
-  convertResult: ConvertResult;
-} & {
-  patient: Pick<Patient, "data" | "id">;
-}): DocumentQueryProgress => {
-  const docQueryProgress = patient.data.documentQueryProgress ?? {};
-
-  const totalToConvert = docQueryProgress?.convert?.total ?? 0;
-
-  const successfulConvert = docQueryProgress?.convert?.successful ?? 0;
-  const successful = convertResult === "success" ? successfulConvert + 1 : successfulConvert;
-
-  const errorsConvert = docQueryProgress?.convert?.errors ?? 0;
-  const errors = convertResult === "failed" ? errorsConvert + 1 : errorsConvert;
-
-  const status = getStatusFromProgress({ successful, errors, total: totalToConvert });
-
-  docQueryProgress.convert = {
-    ...docQueryProgress?.convert,
-    status,
-    successful,
-    errors,
-  };
-
-  return docQueryProgress;
-};
-
 export function getStatusFromProgress(
   progress: Pick<Progress, "errors" | "successful" | "total">
 ): DocumentQueryStatus {
   const { successful, errors, total } = progress;
   const isConversionCompleted = (successful ?? 0) + (errors ?? 0) >= (total ?? 0);
   return isConversionCompleted ? "completed" : "processing";
 }
+
+export function isProcessingStatus(status?: DocumentQueryStatus | undefined) {
+  if (!status) return false;
+  return status === "processing";
+}
+export function isFinalStatus(status?: DocumentQueryStatus | undefined) {
+  return !isProcessingStatus(status);
+}
+
+export function isProcessing(progress?: Progress | undefined) {
+  return isProcessingStatus(progress?.status);
+}
```

Comment:
This was spread out in diff places - the original reason for this PR is a good example as to why this is a bad idea, so consolidated it here and update other places to use it.